src/Disks/ObjectStorages/Web/MetadataStorageFromStaticFilesWebServer.cpp

```diff
@@ -124,39 +123,21 @@ std::vector<std::string> MetadataStorageFromStaticFilesWebServer::listDirectory(
     return result;
 }
 
-bool MetadataStorageFromStaticFilesWebServer::initializeIfNeeded(const std::string & path, std::optional<bool> throw_on_error) const
+void MetadataStorageFromStaticFilesWebServer::initializeIfNeeded(const std::string & path) const
 {
     if (object_storage.files.find(path) == object_storage.files.end())
     {
-        try
-        {
-            object_storage.initialize(fs::path(object_storage.url) / path);
-        }
-        catch (...)
-        {
-            const auto message = getCurrentExceptionMessage(false);
-            bool can_throw = throw_on_error.has_value() ? *throw_on_error : CurrentThread::isInitialized() && CurrentThread::get().getQueryContext();
-            if (can_throw)
-                throw Exception(ErrorCodes::NETWORK_ERROR, "Cannot load disk metadata. Error: {}", message);
-
-            LOG_TRACE(&Poco::Logger::get("DiskWeb"), "Cannot load disk metadata. Error: {}", message);
-            return false;
-        }
+        object_storage.initialize(fs::path(object_storage.url) / path);
```

Comment:
hm, I thought it will throw on select query when calling iterate directory as it also tries to initialize metadata for the path at that point. Looks like query context is not used in this case even if there is a query?

Reply:
The problem here is that `exists()` does not check return value of `initialiizeIfNeeded` and trying to use invalid iterator, OK, let's fix it in a more conservative way...

Reply:
But it still looks pretty tricky, especially in case of some caches and so on, plus if you will forget to attach the query it will also ignore the exception, looks unsafe. I guess since it is only for `exists` it is more or less fine...

So what do you think about removing this `throw_on_error` (and fix tests)?
I doubt that this disk is used in a real big production, so it should not affect lots of users.

Reply:
>The problem here is that exists() does not check return value of initialiizeIfNeeded and trying to use invalid iterator, OK, let's fix it in a more conservative way...

I've mixed here another issue, please ignore this comment.

The problem is way more obvious - server simply does not found any parts on attach, and that's why it simply returns no rows.

Reply:
> The problem is way more obvious - server simply does not found any parts on attach, and that's why it simply returns no rows.

so this number is saved somewhere on attach and on later queries we just do not even check the disk because assume there are no parts?

Reply:
When MergeTree table loads it scans disks for data parts, and in case of HTTP server is not available it cannot find any, so it simply loads table with 0 parts.

Reply:
Ok, agree, we should not allow that.

Reply:
But let's add retries for initialise then? Not it does not retry

Reply:
>though http buffer has retries...

If only they were used... Fixed in 522d37fdd3f11768002cf4c40a39ca6b148ec79a